### DISCUSSION TITLE
pkexec returns "error getting authority" (Kali Linux, WSL 2)

### DISCUSSION THREAD

**Discussion by LRitzdorf** [microsoft/WSL#6178] [General] [answered]
# Environment

```none
Windows build number: 19041.572
Your Distribution version: Kali Rolling (currently on 2020.3)
Whether the issue is on WSL 2 and/or WSL 1: WSL 2
```

# Steps to reproduce

The `pkexec` command, when used to run at least some other commands, returns `Error getting authority: Error initializing authority: Could not connect: No such file or directory`. This was discovered using Wireshark in Kali Linux (on WSL 2) -- the "start menu" shortcut for Wireshark runs `pkexec wireshark`, and does not function. However, running `sudo wireshark` from the command line works as expected, so the issue does not lie with Wireshark. For ease of testing, the error also appears with `pkexec ls`.

This was tested in Kali Linux, and has not yet been verified for other WSL distributions.

#  Expected behavior

```
$ pkexec wireshark
[Wireshark starts, printing status messages in the process]
```

# Actual behavior

```
$ pkexec wireshark
Error getting authority: Error initializing authority: Could not connect: No such file or directory
```

**Comment by therealkenc** [comment, marked as answer]
You don't have `dbus` running. You can do a `sudo service dbus start` to get one step of the way, but there would be more to it to get [polkit](https://gitlab.freedesktop.org/polkit/polkit/) running on a headless Linux box such as WSL. Spiritually dupe #6117.

**Reply by LRitzdorf** [reply]
Thanks, I'll add my observation about pkexec onto #6117 then.

Also, since I'm using Win-KeX (which gives me a GUI using VNC), does that still count as headless for polkit purposes? Plus, I would think that polkit shouldn't be an issue anyway, since I ran `sudo apt install kali-linux-default` to install all the standard tools. Does polkit usually require extra setup like this?

**Comment by moritzbeck13** [comment]
On Kali, running Wireshark from the application menu is linked to `sh -c "$(id -Gn | grep -q wireshark || echo pkexec) wireshark %f"`
On WSL, the command behind the start menu entry is `sh -c "$(id -Gn | grep -q wireshark || echo pkexec) wireshark`

From running this command in PowerShell, I can see two things:
- the quotes are not properly closed
- PowerShell complains about the ||

After starting dbus, which isn't in systemd on WSL for some reason, and running `sh -c "$(id -Gn | grep -q wireshark | echo pkexec) wireshark"`, PowerShell still complains, but Wireshark starts as expected.
I think WSL has some problems interpreting the %f and just cuts of the string altogether. Not sure about the rest.

Funnily enough; this was working for me with an existing install but recent version of Windows 11, but not on a fresh install.

**Reply by LRitzdorf** [reply]
Since asking the initial question here, I've switched to Linux as my primary OS, so I can't examine this myself at the moment. However, the fact that it's cut off at the `%f` feels like maybe some higher level of parsing is eating that token, so the full command doesn't make it through to the shortcut's command field? (This is a wild guess with no real basis; just my initial reaction.)
